TLM/TLM/Manager/Impl/ExtPathManager.cs

```diff
@@ -344,7 +349,12 @@ public class ExtPathManager
                 return true;
             }
 
-            LoopUtil.SpiralLoop(centerI, centerJ, width, height, FindHelper);
+            for (int i = 0; i < _spiralGridCoordsCache.Length; i++) {
```

Comment:
Similar to code block in `AdvancedParkingManager` - maybe split out in to some sort of spiral loop util class?

Reply:
Though about that during the implementation but decided against it since this is perf critical code and I don't see a way to do what you say without additional mem allocations. I could be wrong though, maybe you have a suggestion of how the implementation could look like?